assets/app/scripts/directives/podChart.js

```diff
@@ -0,0 +1,123 @@
+"use strict";
+
+angular.module('openshiftConsole')
+  .directive('podChart', function($timeout, hashSizeFilter, isTroubledPodFilter, Logger) {
+    // Make sure our charts always have unique IDs even if the same deployment
+    // or monopod is shown on the overview more than once.
+    var lastId = 0;
+
+    return {
+      restrict: 'E',
+      scope: {
+        pods: '=',
+        desired: '=?'
+      },
+      templateUrl: 'views/_pod-chart.html',
+      link: function($scope, element) {
+        // The phases to show (in order).
+        var phases = ["Running", "Warning", "Failed", "Pending", "Succeeded", "Unknown"];
+
+        lastId++;
+        $scope.chartId = 'pods-donut-chart-' + lastId;
+
+        // c3.js config for the pods donut chart
+        $scope.chartConfig = {
+          type: "donut",
+          donut: {
+            width: 8,
+            label: {
+              show: false
+            }
+          },
+          size: {
+            height: 130,
+            width: 130
+          },
+          legend: {
+            show: false
+          },
+          tooltip: {
+            format: {
+              value: function(value) {
+                return value;
+              }
+            },
+            position: function() {
+              // Position in the top-left to avoid problems with tooltip text wrapping.
+              return { top: 0, left: 0 };
+            }
+          },
+          data: {
+            type: "donut",
+            groups: [ phases ],
+            // Keep groups in our order.
+            order: null,
+            colors: {
+             Running: "#6eb664",
```

Comment:
fine inline for now but if we start copying this around at all we should probably put these in one place